src/Playwright/Core/Waiter.cs

```diff
@@ -159,6 +166,10 @@ internal async Task<T> WaitForPromiseAsync<T>(Task<T> task, Action dispose = nul
         {
             try
             {
+                if (_immediateError != null)
+                {
+                    throw _immediateError;
```

Comment:
I think we should do `throw new PlaywrightException('some message', _immediateError);` so we keep the stack trace.